app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -231,12 +232,26 @@ class DartdocJobProcessor extends JobProcessor {
     double coverageScore = 0.0;
     if (hasContent && dartdocData != null) {
       final total = dartdocData.apiElements.length;
-      final documented = dartdocData.apiElements
-          .where((elem) =>
-              elem.documentation != null &&
-              elem.documentation.isNotEmpty &&
-              elem.documentation.trim().length >= 5)
-          .length;
+
+      bool isDocumented(ApiElement elem) {
```

Comment:
Somewhat unrelated... but since we starting to grow the complexity around detecting documentation level..
How about we move all the logic into a function in a separate file... I mean isn't it enough to pass `apiElements` and have it return `Suggestion` or `null`...

```dart
/// Returns a [Suggestion] or `null`, if the documentation level is sufficient.
Suggestion documentationSuggestion(apiElements) {
  ...
}
```

Reply:
With the other PR to make non-linear scores this code grows even more.